app/components/directives/teacher-modal.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Created by maddog on 3/4/2018.
+ */
+
+(
+    function(){
+        angular.module('MockApp')
+            .directive('teacherModal',teacherModal);
+
+        function teacherModal(){
+            var directive = {
+                restrict: 'EA',
+                scope:{
+                    'type': '@modalType',
+                    'title': '@buttonTitle',
+                    'class': '@buttonClass',
+                    'fileName': '@fileName',
+                    modalValue: '=',
+                },
+                template:'<a type="button" class="{{class}}" ng-click="modalDCtrl.open(modalValue,fileName)">{{title}}</a>',
+                controller:'modalDirectiveController as modalDCtrl',
+            }
+            return directive;
+        }
+
+        angular.module('MockApp')
+            .controller('modalDirectiveController', modalDirectiveController);
```

Comment:
Please use separate files for different files. Do not use a single file for everything.